### PR TITLE
AIT support to multiple tavern files in one call

### PR DESCRIPTION
**Pull request by mhamra**
|Related issue|
|---|
|#19280|

## Description

The changes implemented in this PR allow running several pytest tavern files in the same command line.

### CODE REVIEW

File: api/test/integration/conftest.py [hunk 1]
```diff
@@ -12,7 +12,9 @@
 import urllib3
 import yaml
 from py.xml import html
+from git.repo import Repo
 
+global env_mode
```
Comment (GGP1) [inline api/test/integration/conftest.py:17]: The `env_mode` variable is initiated below and set in `api_test()` so it's not necessary to declare it here

```suggestion
```

File: api/test/integration/conftest.py [hunk 2]
```diff
@@ -32,8 +34,26 @@
 standalone_env_mode = 'standalone'
 cluster_env_mode = 'cluster'
 
+# parameters for docker environment start and stop
+values_env_build = {
+    'interval': 10,
+    'max_retries': 3,
+}
+
+# parameters for docker environment healthcheck
```
Comment (GGP1) [inline api/test/integration/conftest.py:43]:
```suggestion
# Parameters for docker environment healthcheck
```

File: api/test/integration/conftest.py [hunk 3]
```diff
@@ -32,8 +34,26 @@
 standalone_env_mode = 'standalone'
 cluster_env_mode = 'cluster'
 
+# parameters for docker environment start and stop
```
Comment (GGP1) [inline api/test/integration/conftest.py:37]:
```suggestion
# Parameters for docker environment start and stop
```

File: api/test/integration/conftest.py [hunk 4]
```diff
@@ -32,8 +34,26 @@
...
+}
+
+# parameters for docker environment healthcheck
+values_env_up = {
+    'interval': 1,
+    'max_retries': 360,
+}
+# env_mode : cluster or standalone
+# Indicates the environment to be used in the process.
+env_mode: str = 'cluster'
```
Comment (GGP1) [inline api/test/integration/conftest.py:50]:
```suggestion
env_mode: str = cluster_env_mode
```

**Review by GGP1 — CHANGES_REQUESTED** [review]
I was able to install the dependencies added and execute two tests in the same command with no issues. 

<details><summary>Tests</summary>

```console
(venv) gasti@pop-os:~/work/wazuh/api/test/integration$ pip install gitpython==3.1.31 gitdb==4.0.10 smmap==5.0.0
Collecting gitpython==3.1.31
  Downloading GitPython-3.1.31-py3-none-any.whl (184 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 184.3/184.3 kB 3.2 MB/s eta 0:00:00
Collecting gitdb==4.0.10
  Using cached gitdb-4.0.10-py3-none-any.whl (62 kB)
Collecting smmap==5.0.0
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Installing collected packages: smmap, gitdb, gitpython
Successfully installed gitdb-4.0.10 gitpython-3.1.31 smmap-5.0.0

[notice] A new release of pip is available: 23.0.1 -> 23.2.1
[notice] To update, run: pip install --upgrade pip
(venv) gasti@pop-os:~/work/wazuh/api/test/integration$ pytest -vv test_manager_endpoints.tavern.yaml test_active_response_endpoints.tavern.yaml
==================================================================== test session starts ====================================================================
platform linux -- Python 3.9.16, pytest-7.3.1, pluggy-1.0.0 -- /home/gasti/work/wazuh/venv/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.9.16', 'Platform': 'Linux-6.2.0-76060200-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.0.0'}, 'Plugins': {'asyncio': '0.18.1', 'tavern': '1.23.5', 'trio': '0.7.0', 'html': '2.1.1', 'aiohttp': '1.0.4', 'metadata': '3.0.0'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, trio-0.7.0, html-2.1.1, aiohttp-1.0.4, metadata-3.0.0
asyncio: mode=auto
collected 34 items                                                                                                                                          

test_manager_endpoints.tavern.yaml::GET /manager/status PASSED                                                                                        [  2%]
test_manager_endpoints.tavern.yaml::GET /manager/info PASSED                                                                                          [  5%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[alerts] PASSED                                                               [  8%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[auth] PASSED                                                                 [ 11%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[cis-cat] PASSED                                                              [ 14%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[cluster] PASSED                                                              [ 17%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[command] PASSED                                                              [ 20%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[global] PASSED                                                               [ 23%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[localfile] PASSED                                                            [ 26%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[osquery] PASSED                                                              [ 29%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[remote] PASSED                                                               [ 32%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[rootcheck] PASSED                                                            [ 35%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[ruleset] PASSED                                                              [ 38%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[sca] PASSED                                                                  [ 41%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[syscheck] PASSED                                                             [ 44%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[syscollector] PASSED                                                         [ 47%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[vulnerability-detector] PASSED                                               [ 50%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration PASSED                                                                                 [ 52%]
test_manager_endpoints.tavern.yaml::GET /manager/daemons/stats PASSED                                                                                 [ 55%]
test_manager_endpoints.tavern.yaml::GET /manager/stats PASSED                                                                                         [ 58%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/hourly PASSED                                                                                  [ 61%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/weekly PASSED                                                                                  [ 64%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/analysisd PASSED                                                                               [ 67%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/remoted PASSED                                                                                 [ 70%]
test_manager_endpoints.tavern.yaml::GET /manager/logs PASSED                                                                                          [ 73%]
test_manager_endpoints.tavern.yaml::GET /manager/logs/summary PASSED                                                                                  [ 76%]
test_manager_endpoints.tavern.yaml::GET /manager/api/config PASSED                                                                                    [ 79%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration/validation (OK) PASSED                                                                 [ 82%]
test_manager_endpoints.tavern.yaml::GET /manager/validation (KO) PASSED                                                                               [ 85%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration/{component}/{configuration} PASSED                                                     [ 88%]
test_manager_endpoints.tavern.yaml::PUT /manager/configuration PASSED                                                                                 [ 91%]
test_manager_endpoints.tavern.yaml::PUT /manager/restart PASSED                                                                                       [ 94%]
test_active_response_endpoints.tavern.yaml::PUT /active-response/{:agent_id} PASSED                                                                   [ 97%]
test_active_response_endpoints.tavern.yaml::PUT /active-response PASSED                                                                               [100%]

===================================================================== warnings summary ======================================================================
../../../venv/lib/python3.9/site-packages/_pytest/nodes.py:642
../../../venv/lib/python3.9/site-packages/_pytest/nodes.py:642
  /home/gasti/work/wazuh/venv/lib/python3.9/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================== 34 passed, 2 warnings in 806.42s (0:13:26) =========================================================
```

</details>

<details><summary>Environment</summary>

```console
gasti@pop-os:~/work/wazuh/api/test/integration$ docker ps -a
CONTAINER ID   IMAGE                              COMMAND                  CREATED              STATUS                        PORTS                                           NAMES
e3a381487a67   integration_test_wazuh-agent_old   "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent8-1
e860f63aab8e   integration_test_wazuh-agent_old   "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent7-1
faa92e8e1c37   integration_test_wazuh-agent_old   "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent6-1
486efe42a23c   integration_test_wazuh-agent_old   "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent5-1
f38987edcaf5   integration_test_wazuh-agent       "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent4-1
e1beb860df07   integration_test_wazuh-agent       "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent3-1
dca0257689ce   integration_test_wazuh-agent       "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent2-1
2207f38b8259   integration_test_wazuh-agent       "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-agent1-1
2fedde61055f   integration_test_nginx-lb          "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)   80/tcp                                          env-nginx-lb-1
f385164bed9f   integration_test_wazuh-manager     "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-worker2-1
33f10756bf40   integration_test_wazuh-manager     "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)                                                   env-wazuh-worker1-1
10a9e1c2b610   integration_test_wazuh-manager     "/scripts/entrypoint…"   About a minute ago   Up About a minute (healthy)   0.0.0.0:55000->55000/tcp, :::55000->55000/tcp   env-wazuh-master-1
```

</details>

I left a few minor suggestions. Great job!

File: api/test/integration/conftest.py [hunk 5]
```diff
@@ -580,4 +697,4 @@ def large_event() -> str:
     str
         The larger event.
     """
-    return " ".join(str(i) for i in range(12773))
+    return " ".join(str(i) for i in range(12773))
```
Comment (GGP1) [inline api/test/integration/conftest.py:700]: Add EOF line

**Review by GGP1 — CHANGES_REQUESTED** [review]

**Review by GGP1 — APPROVED** [review]
LGTM!

**Comment by Selutario** [comment]
Since multiple tavern tests can be run using the script below and they may suffer changes in 5.0.0, we won't be merging this for now: 
- https://github.com/wazuh/wazuh/blob/master/api/test/integration/run_tests.py

**Pull request closed without merging**